api/apps/v1alpha1/nemo_evaluator_types.go

```diff
@@ -826,6 +828,71 @@ func (n *NemoEvaluator) GetServiceMonitorAnnotations() map[string]string {
 	return NemoEvaluatorAnnotations
 }
 
+// GetInitContainers returns the init containers for the NemoEvaluator.
+//
+// It creates and returns a slice of corev1.Container.
+// The init containers include a busybox container to wait for Postgres to start,
+// and an evaluator-db-migration container to run the database migration.
+//
+// Returns a slice of corev1.Container.
+func (n *NemoEvaluator) GetInitContainers() []corev1.Container {
+
+	connCmd := fmt.Sprintf(
+		"until nc -z %s %d; do echo \"Waiting for Postgres to start \"; sleep 5; done",
+		n.Spec.DatabaseConfig.Host,
+		n.Spec.DatabaseConfig.Port)
+
+	envVars := []corev1.EnvVar{
+		{
+			Name:  "NAMESPACE",
+			Value: n.Namespace,
+		},
+		{
+			Name:  "ARGO_HOST",
+			Value: n.Spec.ArgoWorkflows.Endpoint,
+		},
+		{
+			Name:  "EVAL_CONTAINER",
+			Value: n.GetImage(),
+		},
+		{
+			Name:  "DATA_STORE_HOST",
+			Value: n.Spec.Datastore.Endpoint,
+		},
+	}
+	// Append the environment variables for Postgres
+	envVars = append(envVars, n.GetPostgresEnv()...)
+
+	return []corev1.Container{
+		{
+			Name:            "wait-for-postgres",
+			Image:           "busybox",
```

Comment:
lets use the same eval container here

Reply:
The eval container doesnt have `nc` command installed

Reply:
Makes sense. Since this is how nemo-eval team has implemented it,I will have to request the them to add `nc` to the eval container.